src/components/settings/PushNotifications/hooks/useNotificationTracking.ts

```diff
@@ -0,0 +1,82 @@
+import { keys as keysFromIndexedDb, update as updateIndexedDb } from 'idb-keyval'
+import { useEffect, useMemo } from 'react'
+
+import {
+  _DEFAULT_WEBHOOK_TRACKING,
+  createNotificationTrackingIndexedDb,
+  parseNotificationTrackingKey,
+} from '@/services/push-notifications/tracking'
+import { trackEvent } from '@/services/analytics'
+import { PUSH_NOTIFICATION_EVENTS } from '@/services/analytics/events/push-notifications'
+import type { NotificationTracking, NotificationTrackingKey } from '@/services/push-notifications/tracking'
+import type { WebhookType } from '@/service-workers/firebase-messaging/webhook-types'
+
+const trackNotificationEvents = (
+  chainId: string,
+  type: WebhookType,
+  tracking: NotificationTracking[NotificationTrackingKey],
+) => {
+  // Shown notifications
+  Array.from({ length: tracking.shown }).forEach(() => {
```

Comment:
Please use a for loop, this is hard to read and creates an unnecessary object.

Reply:
Adjusted in 14353657ff6ef1b8676bf6e59784935bc6b3a955.